doc/sphinx/language/core/records.rst

```diff
@@ -28,7 +28,9 @@ sense, the :cmd:`Record` construction allows defining“signatures”.
    Each :n:`@record_definition` defines a record named by :n:`@ident_decl`.
    The constructor name is given by :n:`@ident`.
    If the constructor name is not specified, then the default name :n:`Build_@ident` is used,
-   where :n:`@ident` is the record name.
+   where :n:`@ident` is the record name. If given, the :n:`as @ident`
+   part specifies the name to use for inhabitants of the record in the
+   type of projections (see below).
```

Comment:
The existing writeup for "Record" leaves much to be desired.

For readability, we generally try to describe syntax elements in the order that they appear in the grammar.  While this command lacks descriptions for most of the syntax, does this really belong in this first paragraph?  (If it's really closely related to the existing sentences, then perhaps the syntax should be more like `Record foo as id := { bar : nat }.`)

Also, perhaps you would provide a short concrete example of how the `as @ident` name appears in "the name to use for inhabitants of the record in the type of projections (see below)"?  Where exactly is the "below"?  The wording has a few too many abstractions strung together--perhaps it can be simplified?  The doc doesn't define "inhabitant"--can this say "instances"?  Nor does the doc define "projection"--could you provide a definition?

Reply:
> The existing writeup for "Record" leaves much to be desired.

I agree that the section could be improved much.

> the syntax should be more like `Record foo as id := { bar : nat }.`

The `as` refers to the inhabitants of the type, so, it should be in the ":=" part.

> Where exactly is the "below"

Below is the description of projections. But I would rather move up the description of projections closer to the description of the syntax.

Reply:
> The existing writeup for "Record" leaves much to be desired.

I made a separate PR #15489 for that purpose. Tell me what you think.